src/components/Table/TableRow.tsx

```diff
@@ -33,30 +37,36 @@ export const TableRow = ({
     if (
       onChange != undefined &&
       selectRows &&
-      variantStandard === Variant.Body
+      variantStandard === Variant.Body &&
+      rowData
     ) {
-      onChange({ selectedValue: value });
+      onChange({ selectedValue: rowData });
     }
   };
 
   const handleEnter = (event: React.KeyboardEvent<HTMLTableRowElement>) => {
-    if ((event.key === 'Enter' || event.key === ' ') && onChange != undefined) {
-      onChange({ selectedValue: value });
+    if (
+      (event.key === 'Enter' || event.key === ' ') &&
+      onChange != undefined &&
+      selectedValue &&
+      rowData
+    ) {
+      onChange({ selectedValue: rowData });
     }
   };
-
   return (
     <SortContext.Provider value={{ selectSort }}>
       <tr
         {...tableRowProps}
         className={cn(
           classes.TableRow,
           {
-            [classes['table-row--selected']]: value === selectedValue,
+            [classes['table-row--selected']]:
+              JSON.stringify(rowData) === JSON.stringify(selectedValue),
             [classes['table-row--body']]:
               variantStandard === Variant.Body &&
               selectRows &&
-              value !== selectedValue,
+              JSON.stringify(rowData) !== JSON.stringify(selectedValue),
```

Comment:
Instead of potentially passing these values through JSON serialization twice each, maybe you could extract this to a variable like `isSelected`?